course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -0,0 +1,237 @@
+        $httpBackend.verifyNoOutstandingExpectation();
+        $httpBackend.verifyNoOutstandingRequest();
+    });
+
+    // DI sanity check
+    it('should inject the providers we requested', function () {
+        [$controller, $rootScope, $routeParams, courseInstances, $compile,
+            djangoUrl, $httpBackend, $window, $log, $sce].forEach(function (thing) {
+            expect(thing).not.toBeUndefined();
+            expect(thing).not.toBeNull();
+        });
+    });
+
+    describe('setup', function () {
+        beforeEach(function () {
+            controller = $controller('DetailsController', {$scope: scope});
+        });
+
+        afterEach(function () {
+            // handle the course instance get from setTitle, so we can always
+            // assert at the end of a test that there's no pending http calls.
+            $httpBackend.expectGET(courseInstanceURL).respond(200, '');
+            $httpBackend.expectGET(peopleURL).respond(200, '');
+            $httpBackend.flush(2);
+        });
+
+        it('should set the course instance id', function () {
+            expect(dc.courseInstanceId).toEqual($routeParams.courseInstanceId);
+        });
+
+    });
+
+    describe('setCourseInstance', function () {
+        var ci;
+        var members;
+        beforeEach(function () {
+            // if we want to pull the instance id from $routeParams, this has
+            // to be in a beforeEach(), can't be in a describe().
+            ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Title',
+                course: {
+                    school_id: 'abc',
+                    registrar_code_display: '2222'
+                }
+            };
+            members = {
+                count: 100
+            };
+        });
+
+        it('should work when CourseInstance is empty', function () {
+            var dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
+            $httpBackend.flush(2);
+            expect(dc.courseInstance['title']).toEqual(ci.title);
+            expect(dc.courseInstance['school']).toEqual(ci.course.school_id.toUpperCase());
+            expect(dc.courseInstance['members']).toEqual(100);
+        });
+
+        it('should work when CourseInstance returns invalid id', function () {
+            controller = $controller('DetailsController', {$scope: scope});
+            ci.course_instance_id = 12345;
+            spyOn($log, 'error');
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
+            $httpBackend.flush(2);
+            expect($log.error).toHaveBeenCalled();
+        });
+    });
+
+    describe('stripQuotes', function () {
```

Comment:
For this method, we don't actually need `ci` or `members`, right?  It takes a string, strips leading/trailing quotes, and returns it.  We do need to instantiate the controller, in order for the method to get defined, but we're not using the data it retrieves.